presto-main/src/test/java/io/prestosql/execution/TestCreateSchemaTask.java

```diff
@@ -0,0 +1,147 @@
+ * limitations under the License.
+ */
+package io.prestosql.execution;
+
+import com.google.common.collect.ImmutableList;
+import io.prestosql.Session;
+import io.prestosql.connector.CatalogName;
+import io.prestosql.metadata.AbstractMockMetadata;
+import io.prestosql.metadata.Catalog;
+import io.prestosql.metadata.CatalogManager;
+import io.prestosql.metadata.SchemaPropertyManager;
+import io.prestosql.security.AllowAllAccessControl;
+import io.prestosql.spi.PrestoException;
+import io.prestosql.spi.connector.CatalogSchemaName;
+import io.prestosql.spi.security.PrestoPrincipal;
+import io.prestosql.sql.tree.CreateSchema;
+import io.prestosql.sql.tree.QualifiedName;
+import io.prestosql.transaction.TransactionManager;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.CopyOnWriteArrayList;
+
+import static io.airlift.concurrent.MoreFutures.getFutureValue;
+import static io.prestosql.spi.StandardErrorCode.ALREADY_EXISTS;
+import static io.prestosql.testing.TestingSession.createBogusTestingCatalog;
+import static io.prestosql.testing.TestingSession.testSessionBuilder;
+import static io.prestosql.transaction.InMemoryTransactionManager.createTestTransactionManager;
+import static java.util.Collections.emptyList;
+import static java.util.Objects.requireNonNull;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertThrows;
+
+@Test(singleThreaded = true)
+public class TestCreateSchemaTask
+{
+    private static final String CATALOG_NAME = "catalog";
+    private Session testSession;
+    TestCreateSchemaTask.MockMetadata metadata;
+
+    @BeforeMethod
+    public void setUp()
+    {
+        CatalogManager catalogManager = new CatalogManager();
+        TransactionManager transactionManager = createTestTransactionManager(catalogManager);
+        Catalog testCatalog = createBogusTestingCatalog(CATALOG_NAME);
+        catalogManager.registerCatalog(testCatalog);
+        SchemaPropertyManager schemaPropertyManager = new SchemaPropertyManager();
+        schemaPropertyManager.addProperties(testCatalog.getConnectorCatalogName(), ImmutableList.of());
+        testSession = testSessionBuilder()
+                .setTransactionId(transactionManager.beginTransaction(false))
+                .build();
+        metadata = new TestCreateSchemaTask.MockMetadata(
+            schemaPropertyManager,
+            testCatalog.getConnectorCatalogName());
+    }
+
+    @Test
+    public void testDuplicatedCreateSchema()
+    {
+        String schemaName = "test_db";
+        CreateSchema statement = new CreateSchema(QualifiedName.of(schemaName), false, ImmutableList.of());
+        getFutureValue(new CreateSchemaTask().internalExecute(statement, metadata, new AllowAllAccessControl(), testSession, emptyList()));
+        assertEquals(metadata.getCreateSchemaCount(), 1);
+        assertThrows(
```

Comment:
Please change it to use AssertJ assertion and validate that exception message actually says about duplicate schema.

Here is an example: https://github.com/prestosql/presto/blob/7d2d9db8130f76a1d9f7814c1ce61d810fdc6864/presto-parquet/src/test/java/io/prestosql/parquet/TestTupleDomainParquetPredicate.java#L137